applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -0,0 +1,131 @@
+{
+    if (rThisVariable == STRAIN) {
+        rValue = mPreviousRelativeDisplacement;
+    } else if (rThisVariable == CAUCHY_STRESS_VECTOR) {
+        rValue = mPreviousTraction;
+    }
+
+    return rValue;
+}
+
+ConstitutiveLaw::StressMeasure GeoIncrementalLinearElasticInterfaceLaw::GetStressMeasure()
+{
+    return ConstitutiveLaw::StressMeasure_Cauchy;
+}
+
+bool GeoIncrementalLinearElasticInterfaceLaw::IsIncremental() { return true; }
+
+void GeoIncrementalLinearElasticInterfaceLaw::InitializeMaterial(const Properties&,
+                                                                 const ConstitutiveLaw::GeometryType&,
+                                                                 const Vector&)
+{
+    mPreviousRelativeDisplacement =
+        HasInitialState() ? GetInitialState().GetInitialStrainVector() : ZeroVector{GetStrainSize()};
+    mPreviousTraction =
+        HasInitialState() ? GetInitialState().GetInitialStressVector() : ZeroVector{GetStrainSize()};
+}
+
+void GeoIncrementalLinearElasticInterfaceLaw::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    rValues.GetStressVector() =
+        mPreviousTraction +
+        prod(MakeConstitutiveMatrix(rValues.GetMaterialProperties()[INTERFACE_NORMAL_STIFFNESS],
+                                    rValues.GetMaterialProperties()[INTERFACE_SHEAR_STIFFNESS]),
+             rValues.GetStrainVector() - mPreviousRelativeDisplacement);
+}
+
+bool GeoIncrementalLinearElasticInterfaceLaw::RequiresInitializeMaterialResponse() { return false; }
+
+void GeoIncrementalLinearElasticInterfaceLaw::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    mPreviousRelativeDisplacement = rValues.GetStrainVector();
+    mPreviousTraction             = rValues.GetStressVector();
+}
+
+int GeoIncrementalLinearElasticInterfaceLaw::Check(const Properties& rMaterialProperties,
+                                                   const ConstitutiveLaw::GeometryType& rElementGeometry,
+                                                   const ProcessInfo& rCurrentProcessInfo) const
+{
+    const auto result = BaseType::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACE_NORMAL_STIFFNESS))
+        << "No interface normal stiffness is defined" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties[INTERFACE_NORMAL_STIFFNESS] > 0.0)
+        << "Interface normal stiffness must be positive, but got "
+        << rMaterialProperties[INTERFACE_NORMAL_STIFFNESS] << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACE_SHEAR_STIFFNESS))
+        << "No interface shear stiffness is defined" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rMaterialProperties[INTERFACE_SHEAR_STIFFNESS] > 0.0)
+        << "Interface shear stiffness must be positive, but got "
+        << rMaterialProperties[INTERFACE_SHEAR_STIFFNESS] << std::endl;
```

Comment:
Just out of curiosity, would it every be reasonable that the stiffnesses depend on something else (and thus become time-dependent)?

Reply:
Stiffness can depend on many things e.g. time, age, temperature, strain, stress etc. That is what you might use when doing computations that involve improving ground stability by grouting.